src/ios/CDVReachabilityManager.m

```diff
@@ -0,0 +1,210 @@
+#import "CDVReachabilityManager.h"
+
+@implementation ProxyInfo
+- (BOOL)isEqual:(id)object {
+    ProxyInfo *toCompare = (ProxyInfo *)object;
+    BOOL isEqual = NO;
+    if (toCompare) {
+        isEqual = [toCompare.type isEqualToString:self.type] && (toCompare.host == self.host || [toCompare.host isEqualToString:self.host]) && (toCompare.port == self.port || [toCompare.port isEqualToString:self.port]) && (toCompare.urlKey == self.urlKey || [toCompare.urlKey isEqualToString:self.urlKey]);
+    }
+    return isEqual;
+}
+@end
+
+@implementation NetworkInfo
+@end
+
+@implementation ReachabilityManager
+
+
+- (void)start:(NSURL*)url :(SPNetworkInfoBlock)completionHandler {
+    self.proxyUrl = url;
+    self.networkInfoBlock = completionHandler;
+    [self startReachabilityListener];
+}
+
+- (void)stop {
+    [self stopReachabilityListener];
+}
+
+- (void)startReachabilityListener {
+    if (self.reachabilityListenerStarted == YES) {
+        return;
+    }
+    self.reachabilityListenerStarted = YES;
+
+    self.reach = [CPReachability reachabilityForInternetConnection];
+
+    __weak ReachabilityManager* weakSelf = self;
+    self.reach.reachableBlock = ^(CPReachability*reach)
+    {
+        NSLog(@"===qqq REACHABLE!");
+        [weakSelf checkNetworkData];
+    };
+
+    self.reach.unreachableBlock = ^(CPReachability*reach)
```

Comment:
Seems like this code is needed only for DEBUG configuration. Better to wrap it into with `#if-#endif`
```
#if DEBUG
...
#endif
```